### PR TITLE
Update P2P messages - gossip light micro blocks

### PR DESCRIPTION
**Pull request by hanssv**

### CODE REVIEW

File: sync/p2p_messages.md [hunk 1]
```diff
@@ -139,6 +150,38 @@ Message is RLP encoded, fields:
...
+A signed transaction is serialized as a tagged and versioned
+[signed transaction](../serializations.md#signed-transaction).
+
+## MSG_KEY_BLOCK
+*(Tag = 10)
+
+Message is RLP encoded, fields:
+  - `KeyBlock :: byte_array - Serialized key block`
+
+The key block is [serialized](../serializations.md#key_block) using the
```
Comment (lucafavatella) [inline sync/p2p_messages.md]: Dead anchor. Anyway, rebasing on current master it shall be `#key-block-header`

Comment (lucafavatella) [reply sync/p2p_messages.md]: And not clear why function named if spec linked.

File: sync/p2p_messages.md [hunk 2]
```diff
@@ -139,6 +150,38 @@ Message is RLP encoded, fields:
...
+`aec_blocks:serialize_to_binary/1` function.
+
+## MSG_MICRO_BLOCK
+*(Tag = 11)
+
+Message is RLP encoded, fields:
+  - `MicroBlock :: byte_array - Serialized micro block`
+  - `Light :: bool - flag if micro block is light or normal`
+
+A normal micro block is [serialized](../serializations.md#micro_block) using
```
Comment (lucafavatella) [inline sync/p2p_messages.md]: Dead anchor.

**Comment by hanssv** [comment]
Rebased and fixed anchors

File: sync/p2p_messages.md [hunk 3]
```diff
@@ -139,6 +150,36 @@ Message is RLP encoded, fields:
...
+The key block is [serialized](../serializations.md#key-block-header).
+
+## MSG_MICRO_BLOCK
+*(Tag = 11)
+
+Message is RLP encoded, fields:
+  - `MicroBlock :: byte_array - Serialized micro block`
+  - `Light :: bool - flag if micro block is light or normal`
+
+A normal micro block is [serialized](../serializations.md#micro-block-header).
```
Comment (lucafavatella) [inline sync/p2p_messages.md]: Linking header here - no txs - though mentioning txs in next sentence? I need to double check.

File: sync/p2p_messages.md [hunk 4]
```diff
@@ -123,6 +127,13 @@ Message is RLP encoded, fields:
 Each header hash contains a 64-bit big endian height and the corresponding
 hash, see `aec_peer_messages` for details.
 
+## MSG_GET_BLOCK_TXS
+*(Tag = 7)*
```
Comment (lucafavatella) [inline sync/p2p_messages.md]: This is correct:
https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/include/aec_peer_messages.hrl#L9

File: sync/p2p_messages.md [hunk 5]
```diff
@@ -123,6 +127,13 @@ Message is RLP encoded, fields:
...
+
+Message is RLP encoded, fields:
```
Comment (lucafavatella) [inline sync/p2p_messages.md]: This is correct:
https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/src/aec_peer_messages.erl#L60

File: sync/p2p_messages.md [hunk 6]
```diff
@@ -139,6 +150,36 @@ Message is RLP encoded, fields:
 A signed transaction is serialized as a tagged and versioned
 [signed transaction](../serializations.md#signed-transaction).
 
+## MSG_BLOCK_TXS
+*(Tag = 13)
```
Comment (lucafavatella) [inline sync/p2p_messages.md]: This is correct:
https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/include/aec_peer_messages.hrl#L12

File: sync/p2p_messages.md [hunk 7]
```diff
@@ -139,6 +150,36 @@ Message is RLP encoded, fields:
...
+
+Message is RLP encoded, fields:
+  - `Hash :: byte_array - The block we fetch TXs from`
+  - `Txs :: [byte_array] - List of serialized signed TXs`
+
+A signed transaction is serialized as a tagged and versioned
+[signed transaction](../serializations.md#signed-transaction).
+
+## MSG_KEY_BLOCK
+*(Tag = 10)
```
Comment (lucafavatella) [inline sync/p2p_messages.md]: This is correct:
https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/include/aec_peer_messages.hrl#L13

File: sync/p2p_messages.md [hunk 8]
```diff
@@ -139,6 +150,36 @@ Message is RLP encoded, fields:
...
+
+Message is RLP encoded, fields:
+  - `KeyBlock :: byte_array - Serialized key block`
+
+The key block is [serialized](../serializations.md#key-block-header).
+
+## MSG_MICRO_BLOCK
+*(Tag = 11)
```
Comment (lucafavatella) [inline sync/p2p_messages.md]: This is correct:
https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/include/aec_peer_messages.hrl#L14

File: sync/p2p_messages.md [hunk 9]
```diff
@@ -139,6 +150,36 @@ Message is RLP encoded, fields:
 A signed transaction is serialized as a tagged and versioned
 [signed transaction](../serializations.md#signed-transaction).
 
+## MSG_BLOCK_TXS
+*(Tag = 13)
+
+Message is RLP encoded, fields:
```
Comment (lucafavatella) [inline sync/p2p_messages.md]: This is correct:
https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/src/aec_peer_connection.erl#L1004-L1005

File: sync/p2p_messages.md [hunk 10]
```diff
@@ -139,6 +150,36 @@ Message is RLP encoded, fields:
...
+
+## MSG_MICRO_BLOCK
+*(Tag = 11)
+
+Message is RLP encoded, fields:
+  - `MicroBlock :: byte_array - Serialized micro block`
+  - `Light :: bool - flag if micro block is light or normal`
+
+A normal micro block is [serialized](../serializations.md#micro-block-header).
+A light micro block is serialized using
```
Comment (lucafavatella) [inline sync/p2p_messages.md]: For light micro block:
* https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/src/aec_peer_connection.erl#L932-L935
* https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/src/aec_peer_connection.erl#L1165-L1176
* https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/src/aec_peer_connection.erl#L959-L960

So description seems ok (even if referring to code).

File: sync/p2p_messages.md (hunk 3 above)
Comment (lucafavatella) [inline sync/p2p_messages.md]: The source for this is
https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/src/aec_peer_connection.erl#L953-L954
that ends up in
https://github.com/aeternity/epoch/blob/c247b32db2116aa9db700da01aeaae1c2944641e/apps/aecore/src/aec_blocks.erl#L338-L343

so correct link seems [`/serializations.md#micro-block`](https://github.com/aeternity/protocol/blob/836975660a6ff94888930f9d82ebb2db1a54ea95/serializations.md#micro-block) ? cc @gorillainduction

Comment (hanssv) [reply sync/p2p_messages.md]: Yes, correct, wrong link.

**Comment by hanssv** [comment]
Links to the correct parts of serialization.md (+ rebase).

**Pull request merged**